#**Proyecto: Casos de violencia familiar y  su relación con el aislamiento por COVID-19 **


###Equipo 23

* Francisco Ricardo Arredondo Almuina - frankk_arredondo@hotmail.com
* Jesus Emanuel Serrano Molina - emanuel.serm@gmail.com
* Carlos Eduardo Vidal Villeda - charlie.lalo@hotmail.com

#  **IDENTIFICACIÓN DEL PROBLEMA**

## Problemáticas

1. Violencia familiar en tiempos de COVID – 19

2. Violencia de género en tiempos de COVID – 19

## Investigación 
- Díaz Olavarrieta, C., & Phillips, V. (s.f.). COVID-19 y Violencia Basada en Género/Violencia Intrafamiliar. Recuperado el 24 de Febrero de 2021, de ACADEMIA NACIONAL DE MEDICINA DE MÉXICO: https://www.anmm.org.mx/descargas/COVID-19-y-Violencia-Basada-en-Genero-Violencia-Intrafamiliar.pdf

- EQUIS Justicia para las Mujeres, l. R. (s.f.). LAS DOS PANDEMIAS: VIOLENCIA CONTRA LAS MUJERES EN MÉXICO EN EL CONTEXTO DEL COVID-19. Recuperado el 24 de Febrero de 2021, de EQUIS Justicia para las Mujeres: https://equis.org.mx/wp-content/uploads/2020/08/informe-dospandemiasmexico.pdf

- Fenández Nicto, B. (3 de Mayo de 2020). Violencia Doméstica en México durante el confinamiento por la pandemia COVID-19. (I. Yánez Soria, & M. Antonia Bravo, Editores) Recuperado el 24 de Febrero de 2021, de DATA-POP ALLIANCE: https://datapopalliance.org/violencia-domestica-en-mexico-durante-el-confinamiento-por-la-pandemia-covid-19/

- Gómez Macfarland, C. A., & Sánchez Ramírez, M. C. (Junio de 2020). Violencia familiar en tiempos de Covid. Recuperado el 24 de Febrero de 2021, de Instituto Belisario Domínguez: http://bibliodigitalibd.senado.gob.mx/bitstream/handle/123456789/4891/ML_187.pdf?sequence=1&isAllowed=y

- Llamas Caballero, R. (s.f.). Violencia doméstica y de género en tiempos de COVID-19. Recuperado el 24 de Febrero de 2021, de Anáhuac México: https://www.anahuac.mx/mexico/noticias/Violencia-domestica-y-de-genero-en-tiempos-de-COVID-19

- Monroy, J. (16 de Abril de 2020). Segob: violencia intrafamiliar aumentó 120% desde la emergencia del Covid-19. Recuperado el 24 de Febrero de 2021, de EL ECONOMISTA: https://www.eleconomista.com.mx/politica/Segob-violencia-intrafamiliar-aumento-120-desde-la-emergencia-del-Covid-19-20200416-0111.html

- Núñez Trejo, A. (7 de Junio de 2020). Crece la violencia intrafamiliar en México, por el COVID-19. Recuperado el 24 de Febrero de 2021, de FORO JURÍDICO: https://forojuridico.mx/crece-la-violencia-intrafamiliar-en-mexico-por-el-covid-19/

- Rodríguez Pedraza, Y. (2020). La feminización de la pandemia COVID19. Recuperado el 24 de Febrero de 2021, de Redalyc: https://www.redalyc.org/articulo.oa?id=29063559023

## Medidas ante las problemáticas 



*   Órdenes de protección.
*   Si identifica que usted experimenta una de las características del ciclo de la violencia, acuda al Agente del Ministerio Público y busque ayuda.
*  Resguarde sus documentos, así como los de sus hijos, en una casa de algún familiar.
*  Si su pareja o el agresor se pone violento, retírese inmediatamente.
*  Acuda a buscar ayuda a las instituciones especializadas (UAVI, Unidad de Atención de Violencia Intrafamiliar, IJM Instituto Jalisciense de las Mujeres, DIF Municipales) para buscar refugio temporal
*  Llame al 066 para pedir ayuda la policía.

#### *Fuentes para medidas ante las problemáticas*

- Hernández, R. (s.f.). MEDIDAS PREVENTIVAS PARA EVITAR LA VIOLENCIA INTRAFAMILIAR. Recuperado el 25 de Febrero de 2021, de FISCALIA GENERAL DEL ESTADO. GOBIERNO DEL ESTADO DE JALISCO: https://fge.jalisco.gob.mx/prevencion-social/medidas-preventivas-para-evitar-la-violencia-intrafamiliar



# **PLANTEAMIENTO DE PREGUNTAS CLAVE**

### 1. ¿Cuáles son los estados con mayor violencia familiar en México?

### 2. ¿Cuáles son los estados con menos incidencia de violencia familiar?

### 3. ¿Cuál es el mes con mayor incidencia de violencia familiar?

### 4. ¿Cuál es el año con mayor incidencia de violencia familiar?

### 5. ¿Es cierto que en el 2020 el crecimiento en incidencia de violencia familiar aumento debido a la pandemia?

### 6. ¿Este fenómeno se repitió en otros países o solo en México?

# **COLECCIÓN DE DATOS**
La colección de datos es una parte fundamental dentro de un proyecto de ciencia de datos, siendo que es importante tener una fuente confiable y segura de los datos que van a ser sometidos a un análisis. Existen distintas fuentes para extraer bases de datos, entre las más destacadas se encuentran sitios como Kaggle, APIs públicas o privadas, Instituciones gubernamentales o privadas, entre otras.

Siendo que nuestro proyecto esta dirigido a analizar la "Violencia intrafamiliar en épocas de Pandemia", es fundamental obtener nuestros datos de una fuente confiable y que realmente nos ayude a poder analizar los delitos cometidos en México.

Para la obtención de los datos en cuanto a la República Mexicana fue usado el website del Gobierno de México, dado que para poder obtener los delitos en cada uno de los Estados y Municipios del país, deben de ser obtenido mediante una entidad gubernamental por temas de confidencialidad y resguardo de datos, por lo que en la siguiente dirección URL se encuentran los datos abiertos de incidencia delictiva en los últimos 5 años:

https://drive.google.com/file/d/1CW1dhLMF0QLZ4vXUTMiBrbj-O53Y9XpL/view?usp=sharing

Cabe destacar que dicha dirección contiene dos archivos .CSV con los distintos delitos cometidos en los diferentes Estados de la República Mexicana, asimismo como en cada Municipio de los mismos, lo cual posteriormente nos brindará un análisis más completo de acuerdo con el fin del proyecto.

Para leer lor achivos mencionados en Python, es posible realizarlo mediante dos maneras, importandolo desde los archivos guardados de la computadora, o previamente haberlo guardado en Google Drive e importarlo directamente a Google Colaboratory. Ambas maneras consiguen obtener los archivos .CSV que servirán para realizar un Análisis Exploratorio, Limpieza y Transformación de nuestros datos.

###**Import from Files**

Primeramente se importaron las librerías necesarias.

In [ ]:
import pandas as pd

Después, tras ya tener los archivos correspondientes en nuestra computadora, correremos el siguiente código para poder elegir estos archivos y que Google colaboratory pueda leerlos y almacenarlos en los archivos de la Notebook.

In [ ]:
from google.colab import files
files.upload()

Tras almacenar los datos dentro de los archivos del Notebook, copiaremos la ruta de los mismos para poder leerlos mediante pandas. Nombrando al dataframe con los datos de delincuencia de los Estados y Municipios de México como "df_estados" y "df_municipios", respectivamente.

Cabe destacar que para que puedan ser leidos los documentos estos deben de tener un encoding de "latin-1" para que python pueda leerlo en base al idioma de los datos. 

In [ ]:
df_estados = pd.read_csv("/content/Estatal-Delitos-2015-2021_ene2021.csv", encoding="latin-1")
df_municipios = pd.read_csv("/content/Municipal-Delitos-2015-2021_ene2021.csv", encoding="latin-1")

Finalmente para ver que nuestro dataframes fueron almacenados correctamente y explorar un poco su información se usó el comando "head"

In [ ]:
df_estados.head(2)

Año  Clave_Ent         Entidad  ... Octubre Noviembre Diciembre
0  2015          1  Aguascalientes  ...       2       2.0       1.0
1  2015          1  Aguascalientes  ...       0       0.0       1.0

[2 rows x 19 columns]

In [ ]:
df_municipios.head(2)

Año  Clave_Ent         Entidad  ...  Octubre Noviembre Diciembre
0  2015          1  Aguascalientes  ...      1.0       0.0       1.0
1  2015          1  Aguascalientes  ...      0.0       0.0       0.0

[2 rows x 21 columns]

###**Import from Drive**

Primeramente se importaron las librerías necesarias, tal cual el caso anterior.

In [3]:
import pandas as pd

A diferencia de importar los archivos desde nuestra computadora, aquí son almacenados previamente desde Google Drive, y mediante el siguiente código podemos traerlos a los archivos de nuestra Notebook. 

Cabe destacar que la dirección de la cual son obtenidos los archivos desde Google Drive varía de acuerdo a donde esté almacenado, por lo que en nuestro caso fue el siguiente.

In [1]:
from google.colab import drive
drive.mount('/content/drive/')
!ls "/content/drive/MyDrive/Data_Proy"

Mounted at /content/drive/
'description analysis exploratorio.aac'
 Estatal-Delitos-2015-2021_ene2021.csv
 Municipal-Delitos-2015-2021_ene2021.csv


Una vez teniendo dichos archivos ya en el Notebook, se realiza la misma secuencia para leer los archivos .CSV mediante pandas, nombrándolos como "df_estados" y "df_municipios".

In [4]:
df_estados = pd.read_csv("/content/drive/MyDrive/Data_Proy/Estatal-Delitos-2015-2021_ene2021.csv", encoding="latin-1")
df_municipios = pd.read_csv("/content/drive/MyDrive/Data_Proy/Municipal-Delitos-2015-2021_ene2021.csv", encoding="latin-1")

Finalmente para ver que nuestro dataframes fueron almacenados correctamente y explorar un poco su información se usó el comando "head"

In [ ]:
df_estados.head(2)

Año  Clave_Ent         Entidad  ... Octubre Noviembre Diciembre
0  2015          1  Aguascalientes  ...       2       2.0       1.0
1  2015          1  Aguascalientes  ...       0       0.0       1.0

[2 rows x 19 columns]

In [ ]:
df_municipios.head(2)

Año  Clave_Ent         Entidad  ...  Octubre Noviembre Diciembre
0  2015          1  Aguascalientes  ...      1.0       0.0       1.0
1  2015          1  Aguascalientes  ...      0.0       0.0       0.0

[2 rows x 21 columns]

# **ANÁLISIS EXPLORATORIO DE DATOS**

Una vez que hemos importado nuestros archivos vamos a empezar a analizar nuestro conjunto de datos usando pandas. Nos ayudamos con algunas preguntas que nos indicarán si nuestra información recolectada nos será de ayuda o no.

1. **¿El conjunto de datos que tengo realmente me sirve para responder algunas de las preguntas que me planteé?**

In [5]:
#previamente se cargaron nuestros archivos csv y se asignaron a las variables df_estados y df_municipios
df_estados = pd.DataFrame(df_estados)
df_estados.head(2)

Año  Clave_Ent         Entidad  ... Octubre Noviembre Diciembre
0  2015          1  Aguascalientes  ...       2       2.0       1.0
1  2015          1  Aguascalientes  ...       0       0.0       1.0

[2 rows x 19 columns]

In [6]:
df_municipios = pd.DataFrame(df_municipios)
df_municipios.head(2)

Año  Clave_Ent         Entidad  ...  Octubre Noviembre Diciembre
0  2015          1  Aguascalientes  ...      1.0       0.0       1.0
1  2015          1  Aguascalientes  ...      0.0       0.0       0.0

[2 rows x 21 columns]

In [ ]:
df_municipios.tail(2)

Año  Clave_Ent    Entidad  ...  Octubre Noviembre Diciembre
1588774  2021         32  Zacatecas  ...      NaN       NaN       NaN
1588775  2021         32  Zacatecas  ...      NaN       NaN       NaN

[2 rows x 21 columns]

_*Con un análisis rápido de nuestros data sets encotramos datos que incluyen información relevante a nuestra investigación, como lo son: tipos de delitos, estado de la República mexicana en donde ocurrieron y fecha en la que ocurrieron.
Por lo tanto podemos ser positivos de que nuestras preguntas serán respondidas*_


2. **¿Qué tamaño tiene mi conjunto de datos? ¿Serán datos suficientes?**

In [ ]:
df_estados.shape

(21952, 19)

In [ ]:
df_municipios.shape

(1588776, 21)

*Vemos que nuestro data frame **`df_estados`** tiene 19 columnas y 21952 filas,, mientras que el data frame **`df_municipios`** cuenta con 21 columnas y 1588776 filas, esta diferencia de filas es posible que se deba a que nuestro segundo data frame agrega la información en donde los actos de violencia fueron registrados (columna de municipios).
Debido a esto y a que ambos data sets fueron obtenidos del mismo lugar, en adelante solo se considerará **`df_municipios`** ya que la información está más completa agragando el campo de los municipios, mientras que **`df_estados`** es una recopilación de estos y estaríamos cometiendo un error de replica de datos*

3. **¿Qué columnas tengo y qué información tengo en cada una de esas columnas?**

In [ ]:
df_municipios.columns

Index(['Año', 'Clave_Ent', 'Entidad', 'Cve. Municipio', 'Municipio',
       'Bien jurídico afectado', 'Tipo de delito', 'Subtipo de delito',
       'Modalidad', 'Enero', 'Febrero', 'Marzo', 'Abril', 'Mayo', 'Junio',
       'Julio', 'Agosto', 'Septiembre', 'Octubre', 'Noviembre', 'Diciembre'],
      dtype='object')

- En la columna **`Año`** tenemos registros que empiezan en el 2015 y llegan hasta el 2021.
- En **`Entidad`** yacen los nombres de los estados donde ocurren los hechos mientras que **`Caleve_Ent`** es la calve federativa de este estado.
- Así en **`Municipio`** se agregan el municipio del estado correspondiente y **`Cve. Municipio`** es su clave única.
- **`Bien jurídico afectado`** , **`Tipo de delito`**, **`Subtipo de delito`** y **`Modalidad`** son columnas que describen los delitos registrados.
- Las columnas de **`Enero`**, **`Febrero`**, **`Marzo`**, **`Abril`**, **`Mayo`**, **`Junio`**, **`Julio`**, **`Agosto`**, **`Septiembre`**, **`Octubre`**, **`Noviembre`**, **`Diciembre`** son las sumas acumuladas de los registros por tipos de delito.




####-
4. **Los nombres que tienen mis columnas, ¿son el nombre más apropiado?** 

- *Si, son descriptivas, simples y van de acuerdo a a la información que queremos obtener*

####-
5. **¿Qué tipos de datos tengo en cada columna? ¿Parecen ser el tipo correcto de datos? ¿O es un tipo de datos "incorrecto"?** 

In [ ]:
df_municipios.dtypes

Año                         int64
Clave_Ent                   int64
Entidad                    object
Cve. Municipio              int64
Municipio                  object
Bien jurídico afectado     object
Tipo de delito             object
Subtipo de delito          object
Modalidad                  object
Enero                       int64
Febrero                   float64
Marzo                     float64
Abril                     float64
Mayo                      float64
Junio                     float64
Julio                     float64
Agosto                    float64
Septiembre                float64
Octubre                   float64
Noviembre                 float64
Diciembre                 float64
dtype: object

- *Las columnas tienen el tipo de dato correcto, especialmente nos interesa que las columnas de fecha como `Año` y los casos acumulados sean datos de tipo númerico.*

6. **Si selecciono algunas filas al azar y las observo, ¿estoy obteniendo los datos que debería? ¿o hay datos que parecen estar "sucios" o "incorrectos"?** 

In [ ]:
df_municipios.loc[[0,182,1112,45932,1588775], ['Año', 'Entidad', 'Municipio', 'Tipo de delito','Modalidad', 'Enero' ]]

Año  ... Enero
0        2015  ...     2
182      2015  ...     0
1112     2015  ...     4
45932    2015  ...     0
1588775  2021  ...     0

[5 rows x 6 columns]

- *Hasta el momento todo parece ser correcto, los datos no muestran estar sucios*

# **LIMPIEZA DE DATOS**



In [ ]:
#Este el el dataframe de los datos organizados por municipio
df_municipios

Año  Clave_Ent         Entidad  ...  Octubre Noviembre Diciembre
0        2015          1  Aguascalientes  ...      1.0       0.0       1.0
1        2015          1  Aguascalientes  ...      0.0       0.0       0.0
2        2015          1  Aguascalientes  ...      0.0       0.0       0.0
3        2015          1  Aguascalientes  ...      0.0       0.0       0.0
4        2015          1  Aguascalientes  ...      0.0       0.0       0.0
...       ...        ...             ...  ...      ...       ...       ...
1588771  2021         32       Zacatecas  ...      NaN       NaN       NaN
1588772  2021         32       Zacatecas  ...      NaN       NaN       NaN
1588773  2021         32       Zacatecas  ...      NaN       NaN       NaN
1588774  2021         32       Zacatecas  ...      NaN       NaN       NaN
1588775  2021         32       Zacatecas  ...      NaN       NaN       NaN

[1588776 rows x 21 columns]

In [7]:
#En este bloque se analiza si existe prescencias de nans en nuetsro dataframe
df_municipios.isna().sum(axis=0)

Año                            0
Clave_Ent                      0
Entidad                        0
Cve. Municipio                 0
Municipio                      0
Bien jurídico afectado         0
Tipo de delito                 0
Subtipo de delito              0
Modalidad                      0
Enero                          0
Febrero                   242158
Marzo                     242158
Abril                     242158
Mayo                      242158
Junio                     242158
Julio                     242158
Agosto                    242158
Septiembre                242158
Octubre                   242158
Noviembre                 242158
Diciembre                 242158
dtype: int64

In [8]:
#En este bloque se elimina los nans
df_municipioslimpio= df_municipios.dropna(axis=0, how='any')

In [9]:
#Aqui se corrobora si el codigo anterior se ejecuto de manera correcta
df_municipioslimpio.isna().sum(axis=0)

Año                       0
Clave_Ent                 0
Entidad                   0
Cve. Municipio            0
Municipio                 0
Bien jurídico afectado    0
Tipo de delito            0
Subtipo de delito         0
Modalidad                 0
Enero                     0
Febrero                   0
Marzo                     0
Abril                     0
Mayo                      0
Junio                     0
Julio                     0
Agosto                    0
Septiembre                0
Octubre                   0
Noviembre                 0
Diciembre                 0
dtype: int64

In [10]:
df_municipioslimpio

Año  Clave_Ent         Entidad  ...  Octubre Noviembre Diciembre
0        2015          1  Aguascalientes  ...      1.0       0.0       1.0
1        2015          1  Aguascalientes  ...      0.0       0.0       0.0
2        2015          1  Aguascalientes  ...      0.0       0.0       0.0
3        2015          1  Aguascalientes  ...      0.0       0.0       0.0
4        2015          1  Aguascalientes  ...      0.0       0.0       0.0
...       ...        ...             ...  ...      ...       ...       ...
1346613  2020         32       Zacatecas  ...      0.0       0.0       0.0
1346614  2020         32       Zacatecas  ...      0.0       0.0       0.0
1346615  2020         32       Zacatecas  ...      0.0       0.0       0.0
1346616  2020         32       Zacatecas  ...      0.0       0.0       0.0
1346617  2020         32       Zacatecas  ...      1.0       0.0       1.0

[1346618 rows x 21 columns]

In [11]:
#En esta parte scaamos la suma por mes de todos los años
mes=['Enero', 'Febrero', 'Marzo', 'Abril', 'Mayo', 'Junio', 'Julio', 'Agosto', 'Septiembre','Octubre','Noviembre','Diciembre']
for i in range(0,11):
  print('La suma del mes de ', mes[i], ' = ',  df_municipioslimpio[mes[i]].sum())

La suma del mes de  Enero  =  894267
La suma del mes de  Febrero  =  896080.0
La suma del mes de  Marzo  =  970568.0
La suma del mes de  Abril  =  902224.0
La suma del mes de  Mayo  =  951787.0
La suma del mes de  Junio  =  949497.0
La suma del mes de  Julio  =  948792.0
La suma del mes de  Agosto  =  973212.0
La suma del mes de  Septiembre  =  950207.0
La suma del mes de  Octubre  =  1001677.0
La suma del mes de  Noviembre  =  928524.0


In [12]:
#En esta parte sacamos el minimo por mes de todos los años
for i in range(0,11):
  print('El min del mes de ', mes[i], ' =',  df_municipioslimpio[mes[i]].min())

El min del mes de  Enero  = 0
El min del mes de  Febrero  = 0.0
El min del mes de  Marzo  = 0.0
El min del mes de  Abril  = 0.0
El min del mes de  Mayo  = -1.0
El min del mes de  Junio  = 0.0
El min del mes de  Julio  = 0.0
El min del mes de  Agosto  = 0.0
El min del mes de  Septiembre  = -1.0
El min del mes de  Octubre  = 0.0
El min del mes de  Noviembre  = 0.0


In [ ]:
#Nos percatamos que habia un error en nuestro dataframe donde habia un -1 cuando
#suponemos que tendria que ser 1 por ende tenemos que limpiar los datos nuevamente

In [13]:
df_municipioslimpio['Mayo']=df_municipioslimpio['Mayo'].replace([-1],1)
df_municipioslimpio['Septiembre']=df_municipioslimpio['Septiembre'].replace([-1],1)

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [14]:
#En esta parte sacamos el minimo por mes de todos los años
for i in range(0,11):
  print('El min del mes de ', mes[i], ' =',  df_municipioslimpio[mes[i]].min())

El min del mes de  Enero  = 0
El min del mes de  Febrero  = 0.0
El min del mes de  Marzo  = 0.0
El min del mes de  Abril  = 0.0
El min del mes de  Mayo  = 0.0
El min del mes de  Junio  = 0.0
El min del mes de  Julio  = 0.0
El min del mes de  Agosto  = 0.0
El min del mes de  Septiembre  = 0.0
El min del mes de  Octubre  = 0.0
El min del mes de  Noviembre  = 0.0


In [15]:
#En esta parte sacamos el maximo por mes de todos los años
for i in range(0,11):
  print('El max del mes de ', mes[i], ' =',  df_municipioslimpio[mes[i]].max())

El max del mes de  Enero  = 1146
El max del mes de  Febrero  = 1212.0
El max del mes de  Marzo  = 1471.0
El max del mes de  Abril  = 1445.0
El max del mes de  Mayo  = 1444.0
El max del mes de  Junio  = 958.0
El max del mes de  Julio  = 1219.0
El max del mes de  Agosto  = 1140.0
El max del mes de  Septiembre  = 1099.0
El max del mes de  Octubre  = 1023.0
El max del mes de  Noviembre  = 1156.0


# **TRANSFORMACIÓN DE DATOS**